mlos_core/mlos_core/optimizers/optimizer.py

```diff
@@ -281,9 +259,11 @@ def get_best_observations(
         """
         if len(self._observations) == 0:
             raise ValueError("No observations registered yet.")
-        (configs, scores, contexts) = self.get_observations()
-        idx = scores.nsmallest(n_max, columns=self._optimization_targets, keep="first").index
-        return (configs.loc[idx], scores.loc[idx], None if contexts is None else contexts.loc[idx])
+        observations = self.get_observations()
+        idx = observations.score.nsmallest(
+            n_max, columns=self._optimization_targets, keep="first"
```

Comment:
```suggestion
            n_max, columns=self._optimization_targets, keep="first",
```

trailing comma, format fixups